sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuTaskMetrics.scala

```diff
@@ -78,6 +78,35 @@ class NanoSecondAccumulator extends AccumulatorV2[jl.Long, NanoTime] {
   override def value: NanoTime = NanoTime(_sum)
 }
 
+class WatermarkAccumulator extends AccumulatorV2[jl.Long, Long] {
```

Comment:
nit: can we call this a HighWatermarkAccumulator? Just because I could see in the future us wanting a metric for a min value seen over a window.